src/Umbraco.Infrastructure/PropertyEditors/PropertyEditorsComponent.cs

```diff
@@ -27,32 +32,48 @@ public void Initialize()
         }
 
         public void Terminate()
-        { }
+        {
+            foreach (var t in _terminate) t();
```

Comment:
I like this pattern, even that it is not perfect. IMO it would be better to have a method taking both subscribe and unsubscribe actions, to ensure you always unsubscribe also.

We could then handle the `foreach` in a base class for both a `_terminateActions` and a `_initializeActions`